sidechain/consensus/aura/src/test/mocks/block_importer_mock.rs

```diff
@@ -0,0 +1,144 @@
+/*
+	Copyright 2021 Integritee AG and Supercomputing Systems AG
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+
+*/
+
+use crate::AuraVerifier;
+use core::marker::PhantomData;
+use itp_sgx_crypto::aes::Aes;
+use itp_test::mock::onchain_mock::OnchainMock;
+use itp_types::H256;
+use its_consensus_common::{BlockImport, Error, Result};
+use its_primitives::traits::{ShardIdentifierFor, SignedBlock as SignedSidechainBlockTrait};
+use its_state::SidechainDB;
+use sgx_externalities::SgxExternalities;
+use sp_core::Pair;
+use sp_runtime::traits::Block as ParentchainBlockTrait;
+use std::sync::RwLock;
+
+/// Block importer mock.
+pub struct BlockImportMock<ParentchainBlock, SignedSidechainBlock> {
+	import_result: Option<Result<()>>,
+	imported_blocks: RwLock<Vec<SignedSidechainBlock>>,
+	_phantom: PhantomData<(ParentchainBlock, SignedSidechainBlock)>,
+}
+
+impl<ParentchainBlock, SignedSidechainBlock> BlockImportMock<ParentchainBlock, SignedSidechainBlock>
+where
+	ParentchainBlock: ParentchainBlockTrait<Hash = H256>,
+	SignedSidechainBlock:
+		SignedSidechainBlockTrait<Public = <sp_core::ed25519::Pair as Pair>::Public> + 'static,
+{
+	#[allow(unused)]
```

Comment:
Is this truly unused, or only used with `feature = test`? If it is only feature gated, I'd prefer to just use a compiler flag because no warnings are generated for unused code (at least for clippy, I am not sure about standard compilation warnings).

Reply:
They're used now - and are in the `#[cfg(test)]` feature anyway, so I'll just remove the annotation again.